lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp

```diff
@@ -1472,4 +1472,99 @@ void mlir::torch::onnx_c::populateDefaultDomainAtoF(
+        Value zero = rewriter.create<Torch::ConstantIntOp>(
+            binder.getLoc(), rewriter.getType<Torch::IntType>(),
+            rewriter.getIntegerAttr(rewriter.getIntegerType(64), 0));
+        for (int i = 0; i < shapeSizes[0]; i++) {
+          Value selectIndex = rewriter.create<Torch::ConstantIntOp>(
+              binder.getLoc(), rewriter.getType<Torch::IntType>(),
+              rewriter.getIntegerAttr(rewriter.getIntegerType(64), i));
+          Value extract = rewriter.create<Torch::AtenSelectIntOp>(
+              binder.getLoc(), selectResultType, shape, zero, selectIndex);
+          Value dim = rewriter.create<Torch::AtenItemOp>(
+              binder.getLoc(), rewriter.getType<Torch::IntType>(), extract);
+          dimList.push_back(dim);
+        }
+        Value dimValueList = rewriter.create<Torch::PrimListConstructOp>(
+            binder.getLoc(),
+            Torch::ListType::get(Torch::IntType::get(binder.op->getContext())),
+            dimList);
+        Value noneVal = rewriter.create<Torch::ConstantNoneOp>(binder.getLoc());
+
+        // Get fill_value if it is present.
+        Value value_const;
+        auto attr = binder.op->getAttr("torch.onnx.value");
+        auto resultDType = resultType.getDtype();
+        if (!attr) {
+          if (!resultDType.isa<FloatType>()) {
+            return rewriter.notifyMatchFailure(
+                binder.op,
+                "`value` attr and result type (FloatType) mismatch.");
+          }
+          value_const = rewriter.create<Torch::ConstantFloatOp>(
+              binder.getLoc(), rewriter.getF64FloatAttr(0.0));
+        } else if (attr.isa<mlir::DenseElementsAttr>()) {
+          auto denseAttr = attr.dyn_cast_or_null<DenseElementsAttr>();
+          auto denseAttrEleType = denseAttr.getElementType();
+          if (resultDType.isa<mlir::FloatType>()) {
+            if (!isa<FloatType>(denseAttrEleType)) {
+              return rewriter.notifyMatchFailure(
+                  binder.op,
+                  "`value` attr and result type (FloatType) mismatch.");
+            }
+            float floatValue = denseAttr.getSplatValue<FloatAttr>()
+                                   .getValue()
+                                   .convertToFloat();
+            value_const = rewriter.create<Torch::ConstantFloatOp>(
+                binder.getLoc(), rewriter.getF64FloatAttr(floatValue));
+          } else if (resultDType.isa<mlir::IntegerType>()) {
+            if (!isa<IntegerType>(denseAttrEleType)) {
+              return rewriter.notifyMatchFailure(
+                  binder.op,
+                  "`value` attr and result type (IntegerType) mismatch.");
+            }
+            int64_t intValue = denseAttr.getSplatValue<IntegerAttr>().getSInt();
+            value_const = rewriter.create<Torch::ConstantIntOp>(
+                binder.getLoc(), rewriter.getI64IntegerAttr(intValue));
+          } else {
```

Comment:
If you use the approach above you can handle the `value_const = ...` cases, then just check that `value_const` was set to something without using an `else`

Reply:
Got rid of the else part and decoupled element dtype discovery from constant op creation step.